components/habits/HabitDailyView.tsx

```diff
@@ -0,0 +1,95 @@
+import { css } from '@emotion/react';
+import { useTranslation } from 'next-i18next';
+import Image from 'next/image';
+import { HabitWithHistory } from '../../src/types/habits';
+import CompleteButton from '../entries/CompleteButton';
+import IncompleteButton from '../entries/IncompleteButton';
+import useCompletionPercentages from './hooks';
+
+const styles = {
+  content: css({
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    backgroundColor: '#333',
+    margin: '12px 0px',
+    padding: '12px',
+    gap: '8px',
+  }),
+
+  middleContainer: css({
+    display: 'flex',
+    flexDirection: 'column',
+    textAlign: 'left',
+    minWidth: '30%',
+  }),
+  deadline: css({
+    color: '#91E220',
+  }),
+
+  streakContainer: css({
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+    maxWidth: '100px',
+    padding: '0px 12px',
+  }),
+  streakText: css({
+    color: '#fad15c',
+    paddingLeft: '4px',
+  }),
+};
+
+type HabitDailyViewProps = {
+  habitWithHistory: HabitWithHistory;
+  streak: number;
+  date: string;
+};
+
+const HabitDailyView = ({
+  habitWithHistory,
+  streak,
+  date,
+}: HabitDailyViewProps) => {
+  const { t } = useTranslation(['common', 'habit']);
+  const completionPercentagesByDay = useCompletionPercentages({
+    habitWithHistory,
+    dates: [date],
+  });
+
+  return (
+    <div css={styles.content}>
+      <div css={styles.streakContainer}>
+        <Image
+          src='/images/fire.svg'
+          alt={t('habit:alt.fire')}
+          width='32'
+          height='32'
+        />
+        <span css={styles.streakText}>{streak}</span>
+      </div>
+
+      <div css={styles.middleContainer}>
+        <span>{habitWithHistory.name}</span>
+        <span css={styles.deadline}>
+          {completionPercentagesByDay[date] < 100
+            ? t('common:dayLeft', { quantity: '2' })
+            : t('habit:complete')}
+        </span>
+      </div>
+
+      {/* eslint-disable-next-line no-nested-ternary */}
```

Comment:
Tut tut! the nested ternary rule actually has a point here - the logic below requires a little bit of squinting at ;) 

My comment above will make it easier to handle this (re one component for `CompleteButton`), but you can also extract pieces of this out of the `return` portion and into the body as:

`const SomeC = () => {
// using a function instead of a variable ONLY because we'd like to only evaluate the inside if the condition on L82 is met. Otherwise, i'd declare it as an instance as const completeButton = (completionPercentagesByDay...)
  const getCompleteButton = () => (completionPercentagesByDay[date] === 100 ? <CompleteButton /> : <IncompleteButton />

  return (
   <div>
   ....

  </div>
 )
}